src/Validation/Validator.php

```diff
@@ -806,6 +806,28 @@ public function allowEmptyString($field, $when = true, $message = null)
         return $this->allowEmptyFor($field, self::EMPTY_STRING, $when, $message);
     }
 
+    /**
+     * Requires a field to be not be an empty string.
+     *
+     * Opposite to allowEmptyString()
+     *
+     * @param string $field The name of the field.
+     * @param string|null $message The message to show if the field is empty.
+     * @param bool|string|callable $when Indicates when the field is not allowed
+     *   to be empty. Valid values are false (never), 'create', 'update'. If a
+     *   callable is passed then the field will be required to be not empty when
+     *   the callback returns true.
+     * @return $this
+     * @see \Cake\Validation\Validator::allowEmptyString()
+     * @since 3.8.0
+     */
+    public function notEmptyString($field, $message = null, $when = false)
```

Comment:
```
public function allowEmptyArray($field, $when = true, $message = null)
```

I am a bit concerned about the difference of order here on the public API: when/message vs message/when between the notEmpty and allow methods.